encord/client.py

```diff
@@ -444,6 +453,46 @@ def remove_groups(self, dataset_hash: uuid.UUID, group_hash: List[uuid.UUID]) ->
         params = RemoveGroupsParams(group_hash_list=group_hash)
         self._get_api_client().delete(f"datasets/{dataset_hash}/groups", params=params, result_type=None)
 
+    def __add_data_to_dataset_get_result(
+        self,
+        upload_job_id: str,
+        timeout_seconds: int = 7 * 24 * 60 * 60,  # 7 days
+    ) -> DatasetDataLongPolling:
+        failed_requests_count = 0
+        polling_start_timestamp = time.perf_counter()
+
+        while True:
```

Comment:
maybe `while polling_elapsed_seconds < timeout_seconds:` to be extra safe? (and `return pending` at the end)